vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -251,6 +250,9 @@ func (d *VolumeDriver) Create(r volume.Request) volume.Response {
 	if errAttach != nil {
 		log.WithFields(log.Fields{"name": r.Name,
 			"error": errAttach}).Error("Attach volume failed, removing the volume ")
+		// Ignore error, an internal error in the client may have the volume attached and
+		// must be detached before removing below.
+		d.ops.Detach(r.Name, nil)
```

Comment:
Which internal error in the client ? I assume you refer to an error in d.ops.Attach which may force to return OK but still have the vmdk attached? if there is one,. we need to fix it. The logic "if attach failed, do detach" seems wrong.